src/linalg/impl_linalg.rs

```diff
@@ -532,6 +595,35 @@ fn blas_compat_1d<A, S>(a: &ArrayBase<S, Ix1>) -> bool
 }
 
 #[cfg(feature="blas")]
+fn blas_compat<A, S, D>(a: &ArrayBase<S, D>) -> bool
+    where S: Data,
+          A: 'static,
+          S::Elem: 'static,
+          D: Dimension,
+{
+    if !same_type::<A, S::Elem>() {
+        return false;
+    }
+
+    if a.len() > blas_index::max_value() as usize {
+        return false;
+    }
```

Comment:
Changed it around a bit. I am not sure about whether we should cast to `usize` or `isize` though...